### PR TITLE
Fix SwiftLint rule badge links (#134)

### LINKED ISSUE #134: SwiftLint rule badges do not point to current documentation
It is very helpful that each rule in this style guide links tends to link to a SwiftLint or SwiftFormat rule. However, the SwiftLink links are outdated.

If I click this badge.
Screen Shot 2021-09-08 at 3 25 57 PM

I am taken here.

Screen Shot 2021-09-08 at 3 24 03 PM

It seems like we need to update our SwiftLint links to point to this page: https://realm.github.io/SwiftLint/rule-directory.html.

### PR DESCRIPTION
**Pull request by calda**
This PR updates the SwiftLint rule badge links to point to the current documentation pages.

Fixes #134.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -465,7 +465,7 @@ _You can enable the following settings in Xcode by running [this script](resourc
 
   </details>
 
-* <a id='return-arrow-spacing'></a>(<a href='#return-arrow-spacing'>link</a>) **Place a space on either side of a return arrow for readability.** [![SwiftLint: return_arrow_whitespace](https://img.shields.io/badge/SwiftLint-return__arrow__whitespace-007A87.svg)](https://github.com/realm/SwiftLint/blob/master/Rules.md#returning-whitespace)
+* <a id='return-arrow-spacing'></a>(<a href='#return-arrow-spacing'>link</a>) **Place a space on either side of a return arrow for readability.** [![SwiftLint: return_arrow_whitespace](https://img.shields.io/badge/SwiftLint-return__arrow__whitespace-007A87.svg)](https://realm.github.io/SwiftLint/return_arrow_whitespace)
```
Comment (bachand) [inline README.md:468]: Is it intentional that the slug changed?

File: README.md [hunk 2]
```diff
@@ -1584,7 +1584,7 @@ _You can enable the following settings in Xcode by running [this script](resourc
...
+* <a id='preconditions-and-asserts'></a>(<a href='#preconditions-and-asserts'>link</a>) **Handle an unexpected but recoverable condition with an `assert` method combined with the appropriate logging in production. If the unexpected condition is not recoverable, prefer a `precondition` method or `fatalError()`.** This strikes a balance between crashing and providing insight into unexpected conditions in the wild. Only prefer `fatalError` over a `precondition` method when the failure message is dynamic, since a `precondition` method won't report the message in the crash report. [![SwiftLint: fatal_error_message](https://img.shields.io/badge/SwiftLint-fatal__error__message-007A87.svg)](https://realm.github.io/SwiftLint/fatal_error_message) [![SwiftLint: force_cast](https://img.shields.io/badge/SwiftLint-force__cast-007A87.svg)](https://realm.github.io/SwiftLint/force_cast) [![SwiftLint: force_try](https://img.shields.io/badge/SwiftLint-force__try-007A87.svg)](https://realm.github.io/SwiftLint/force_try) [![SwiftLint: force_unwrapping](https://img.shields.io/badge/SwiftLint-force__unwrapping-007A87.svg)](https://realm.github.io/SwiftLint/force_unwrapping)
```
Comment (bachand) [inline README.md:1587]: On mobile this shows the whole line as changed. I assumed just the last URL did and only reviewed that.

**Review by bachand — APPROVED** [review]
Thanks @calda !

File: README.md (hunk 1 above)
Comment (calda) [reply README.md:468]: https://realm.github.io/SwiftLint/returning_whitespace 404s, the current doc link is https://realm.github.io/SwiftLint/return_arrow_whitespace.

File: README.md (hunk 2 above)
Comment (bachand) [reply README.md:1587]: Looking now on computer I see changed a number of links 👍

**Pull request merged**